examples/oft-solana-v2/tasks/solana/sendOFT.ts

```diff
@@ -0,0 +1,155 @@
+import { OftPDA, oft } from '@layerzerolabs/oft-v2-solana-sdk'
+
+import { createSolanaConnectionFactory } from '../common/utils'
+
+interface Args {
+    amount: number
+    to: string
+    fromEid: EndpointId
+    toEid: EndpointId
+    programId: string
+    mint: string
+    escrow: string
+}
+
+const LOOKUP_TABLE_ADDRESS: Partial<Record<EndpointId, PublicKey>> = {
+    [EndpointId.SOLANA_V2_MAINNET]: new PublicKey('AokBxha6VMLLgf97B5VYHEtqztamWmYERBmmFvjuTzJB'),
+    [EndpointId.SOLANA_V2_TESTNET]: new PublicKey('9thqPdbR27A1yLWw2spwJLySemiGMXxPnEvfmXVk4KuK'),
+}
+
+// Define a Hardhat task for sending OFT from Solana
+task('lz:oft:solana:send', 'Send tokens from Solana to a target EVM chain')
+    .addParam('amount', 'The amount of tokens to send', undefined, types.int)
+    .addParam('fromEid', 'The source endpoint ID', undefined, types.eid)
+    .addParam('to', 'The recipient address on the destination chain')
+    .addParam('toEid', 'The destination endpoint ID', undefined, types.eid)
+    .addParam('mint', 'The OFT token mint public key', undefined, types.string)
+    .addParam('programId', 'The OFT program ID', undefined, types.string)
+    .addParam('escrow', 'The OFT escrow public key', undefined, types.string)
+    .setAction(async (taskArgs: Args) => {
+        const privateKey = process.env.SOLANA_PRIVATE_KEY
+        assert(!!privateKey, 'SOLANA_PRIVATE_KEY is not defined in the environment variables.')
+
+        const keypair = Keypair.fromSecretKey(bs58.decode(privateKey))
+        const umiKeypair = fromWeb3JsKeypair(keypair)
+
+        const lookupTableAddress = LOOKUP_TABLE_ADDRESS[taskArgs.fromEid]
+        assert(lookupTableAddress != null, `No lookup table found for ${formatEid(taskArgs.fromEid)}`)
+
+        const connectionFactory = createSolanaConnectionFactory()
+        const connection = await connectionFactory(taskArgs.fromEid)
+
+        // Initialize Solana connection and UMI framework
+        const umi = createUmi(connection.rpcEndpoint).use(mplToolbox())
+        const umiWalletSigner = createSignerFromKeypair(umi, umiKeypair)
+        umi.use(signerIdentity(umiWalletSigner))
+
+        // Define OFT program and token mint public keys
+        const oftProgramId = publicKey(taskArgs.programId)
+        const umiMintPublicKey = publicKey(taskArgs.mint)
+        const umiEscrowPublicKey = publicKey(taskArgs.escrow)
+
+        // Find the associated token account
+        const tokenAccount = findAssociatedTokenPda(umi, {
+            mint: umiMintPublicKey,
+            owner: umiWalletSigner.publicKey,
+        })
+        console.dir({ tokenAccount })
+
+        // Derive the OFTStore PDA
+        const oftPda = new OftPDA(oftProgramId)
+        const [oftStorePda] = oftPda.oftStore(umiEscrowPublicKey)
+
+        // Fetch token metadata
+        // TODO fix
+        // const mintInfo = (await fetchDigitalAsset(umi, publicKey(tokenAccount[0]))).mint
```

Comment:
For SPL Tokens the way to fetch metadata is pretty straightforward:

```typescript
// Fetch token metadata
const mintInfo = (await fetchDigitalAsset(umi, umiMintPublicKey)).mint
const amount = parseFloat(taskArgs.amount) * 10 ** mintInfo.decimals
```

Not sure if Token2022 is any different.